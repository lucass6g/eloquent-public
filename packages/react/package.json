{
  "name": "@eloquent/react",
  "version": "0.0.1",
  "description": "",
  "main": "./dist/index.js",
  "files": [
    "dist"
  ],
  "types": "./dist/index.d.ts",
  "exports": {
    ".": {
      "import": "./dist/index.js",
      "require": "./dist/index.js",
      "types": "./dist/index.d.ts"
    }
  },
  "scripts": {
    "build": "tsup"
  },
  "keywords": [],
  "author": "",
  "license": "MIT",
  "devDependencies": {
    "tsup": "^8.0.2",
    "@types/react": "^18.2.66",
    "@types/react-dom": "^18.2.22",
    "react": "^18.2.0",
    "typescript": "^5.4.2"
  }
}
