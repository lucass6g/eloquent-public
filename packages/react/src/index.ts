export {Switch, Case, Default} from './components/conditional-switch'