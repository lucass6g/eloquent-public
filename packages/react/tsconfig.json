{
  "extends": "../../packages/config/base.tsconfig.json",
  "include": ["src"],
  "exclude": ["dist", "build", "node_modules"],
  "compilerOptions": {
    "forceConsistentCasingInFileNames": false,
    "rootDir": "src",
    "baseUrl": ".",
    "paths": {
      "@/*": ["./src/*"]
    }
  }
}
