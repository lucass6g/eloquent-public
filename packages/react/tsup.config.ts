import { defineConfig } from 'tsup';

export default defineConfig({
	name: "@eloquent/react",
	entry: ["src/index.ts"],
	outDir: "dist",
	target: "es2022",
	format: ["cjs", "esm"],
	splitting: false,
	sourcemap: true,
	clean: true,
	dts: true,
	external: ["react", 'react-dom', 'react/jsx-runtime'],
	cjsInterop: true,
	treeshake: true,
})