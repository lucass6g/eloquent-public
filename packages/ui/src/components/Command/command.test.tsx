import { render, screen, fireEvent } from '@testing-library/react';
import { describe, it, expect } from 'vitest';
import { Command } from '.';
import { ComponentIcon } from 'lucide-react';


describe('Command', () => {
    it('renders correctly', () => {
        render(<Command.Root>
            <Command.Input placeholder='Digite um comando...' />
            <Command.List>
                <Command.Empty>Sem resultados.</Command.Empty>
                <Command.Group heading="Título da Seção">
                    <Command.Item><ComponentIcon /> Item 1</Command.Item>
                    <Command.Item><ComponentIcon /> Item 2</Command.Item>
                    <Command.Item><ComponentIcon /> Item 3</Command.Item>
                    <Command.Item><ComponentIcon /> Item 4</Command.Item>
                    <Command.Item><ComponentIcon /> Item 5</Command.Item>
                </Command.Group>
            </Command.List>
        </Command.Root>);

        expect(screen.getByPlaceholderText('Digite um comando...')).toBeInTheDocument();
        expect(screen.getByText('Item 1')).toBeInTheDocument();
        expect(screen.getByText('Item 2')).toBeInTheDocument();
    });

    it('show empty result when', () => {
        const component = render(<Command.Root>
            <Command.Input placeholder='Digite um comando...' />
            <Command.List>
                <Command.Empty>Sem resultados.</Command.Empty>
                <Command.Group heading="Título da Seção">
                    <Command.Item><ComponentIcon /> Item 1</Command.Item>
                    <Command.Item><ComponentIcon /> Item 2</Command.Item>
                </Command.Group>
            </Command.List>
        </Command.Root>);

        const input = screen.getByPlaceholderText('Digite um comando...')
        fireEvent.change(input, {
            target: { value: 'xxx' }
        })
        expect(screen.getByText("Sem resultados.")).toBeInTheDocument();

        expect(component.baseElement).toMatchSnapshot();
    });
    it('show correctly result', () => {
        const component = render(<Command.Root>
            <Command.Input placeholder='Digite um comando...' />
            <Command.List>
                <Command.Empty>Sem resultados.</Command.Empty>
                <Command.Group heading="Título da Seção">
                    <Command.Item><ComponentIcon /> Item 1</Command.Item>
                    <Command.Item><ComponentIcon /> Item 2</Command.Item>
                </Command.Group>
            </Command.List>
        </Command.Root>);

        const input = screen.getByPlaceholderText('Digite um comando...')
        fireEvent.change(input, {
            target: { value: 'Item 1' }
        })
        expect(screen.getByText("Item 1")).toBeInTheDocument();

        expect(component.baseElement).toMatchSnapshot();
    });
});