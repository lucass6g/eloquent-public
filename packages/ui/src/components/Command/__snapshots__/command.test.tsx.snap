// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Command > show correctly result 1`] = `
<body>
  <div>
    <div
      class="flex border border-solid border-neutral-300 h-full w-full flex-col overflow-hidden rounded-[8px] bg-popover text-popover-foreground"
      cmdk-root=""
      tabindex="-1"
    >
      <label
        cmdk-label=""
        for=":rj:"
        id=":ri:"
        style="position: absolute; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; border-width: 0px;"
      />
      <div
        class="flex items-center border-b px-3"
        cmdk-input-wrapper=""
      >
        <svg
          class="mr-2 h-4 w-4 shrink-0 opacity-50 text-neutral-500"
          fill="none"
          height="15"
          viewBox="0 0 15 15"
          width="15"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M10 6.5C10 8.433 8.433 10 6.5 10C4.567 10 3 8.433 3 6.5C3 4.567 4.567 3 6.5 3C8.433 3 10 4.567 10 6.5ZM9.30884 10.0159C8.53901 10.6318 7.56251 11 6.5 11C4.01472 11 2 8.98528 2 6.5C2 4.01472 4.01472 2 6.5 2C8.98528 2 11 4.01472 11 6.5C11 7.56251 10.6318 8.53901 10.0159 9.30884L12.8536 12.1464C13.0488 12.3417 13.0488 12.6583 12.8536 12.8536C12.6583 13.0488 12.3417 13.0488 12.1464 12.8536L9.30884 10.0159Z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
        <input
          aria-autocomplete="list"
          aria-controls=":rh:"
          aria-expanded="true"
          aria-labelledby=":ri:"
          autocomplete="off"
          autocorrect="off"
          class="flex h-10 w-full rounded-md bg-transparent py-3 text-sm outline-none placeholder:text-neutral-500 disabled:cursor-not-allowed disabled:opacity-50"
          cmdk-input=""
          id=":rj:"
          placeholder="Digite um comando..."
          role="combobox"
          spellcheck="false"
          type="text"
          value="Item 1"
        />
      </div>
      <div
        aria-label="Suggestions"
        class="max-h-[300px] overflow-y-auto overflow-x-hidden"
        cmdk-list=""
        id=":rh:"
        role="listbox"
      >
        <div
          cmdk-list-sizer=""
        >
          <div
            class="overflow-hidden p-1 text-foreground [&_[cmdk-group-heading]]:px-2 [&_[cmdk-group-heading]]:py-1.5 [&_[cmdk-group-heading]]:text-xs [&_[cmdk-group-heading]]:font-medium [&_[cmdk-group-heading]]:text-neutral-500"
            cmdk-group=""
            data-value="Título da Seção"
            role="presentation"
          >
            <div
              aria-hidden="true"
              cmdk-group-heading=""
              id=":rl:"
            >
              Título da Seção
            </div>
            <div
              aria-labelledby=":rl:"
              cmdk-group-items=""
              role="group"
            >
              <div
                aria-disabled="false"
                aria-selected="true"
                class="relative flex gap-2 cursor-default select-none items-center rounded-[4px] px-2 py-1.5 text-sm outline-none data-[disabled=true]:pointer-events-none data-[selected=true]:bg-neutral-200 data-[selected=true]:text-neutral-800 data-[disabled=true]:opacity-50"
                cmdk-item=""
                data-disabled="false"
                data-selected="true"
                data-value="Item 1"
                id=":rm:"
                role="option"
              >
                <svg
                  class="lucide lucide-component"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M5.5 8.5 9 12l-3.5 3.5L2 12l3.5-3.5Z"
                  />
                  <path
                    d="m12 2 3.5 3.5L12 9 8.5 5.5 12 2Z"
                  />
                  <path
                    d="M18.5 8.5 22 12l-3.5 3.5L15 12l3.5-3.5Z"
                  />
                  <path
                    d="m12 15 3.5 3.5L12 22l-3.5-3.5L12 15Z"
                  />
                </svg>
                 Item 1
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Command > show empty result when 1`] = `
<body>
  <div>
    <div
      class="flex border border-solid border-neutral-300 h-full w-full flex-col overflow-hidden rounded-[8px] bg-popover text-popover-foreground"
      cmdk-root=""
      tabindex="-1"
    >
      <label
        cmdk-label=""
        for=":rc:"
        id=":rb:"
        style="position: absolute; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; border-width: 0px;"
      />
      <div
        class="flex items-center border-b px-3"
        cmdk-input-wrapper=""
      >
        <svg
          class="mr-2 h-4 w-4 shrink-0 opacity-50 text-neutral-500"
          fill="none"
          height="15"
          viewBox="0 0 15 15"
          width="15"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M10 6.5C10 8.433 8.433 10 6.5 10C4.567 10 3 8.433 3 6.5C3 4.567 4.567 3 6.5 3C8.433 3 10 4.567 10 6.5ZM9.30884 10.0159C8.53901 10.6318 7.56251 11 6.5 11C4.01472 11 2 8.98528 2 6.5C2 4.01472 4.01472 2 6.5 2C8.98528 2 11 4.01472 11 6.5C11 7.56251 10.6318 8.53901 10.0159 9.30884L12.8536 12.1464C13.0488 12.3417 13.0488 12.6583 12.8536 12.8536C12.6583 13.0488 12.3417 13.0488 12.1464 12.8536L9.30884 10.0159Z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
        <input
          aria-autocomplete="list"
          aria-controls=":ra:"
          aria-expanded="true"
          aria-labelledby=":rb:"
          autocomplete="off"
          autocorrect="off"
          class="flex h-10 w-full rounded-md bg-transparent py-3 text-sm outline-none placeholder:text-neutral-500 disabled:cursor-not-allowed disabled:opacity-50"
          cmdk-input=""
          id=":rc:"
          placeholder="Digite um comando..."
          role="combobox"
          spellcheck="false"
          type="text"
          value="xxx"
        />
      </div>
      <div
        aria-label="Suggestions"
        class="max-h-[300px] overflow-y-auto overflow-x-hidden"
        cmdk-list=""
        id=":ra:"
        role="listbox"
      >
        <div
          cmdk-list-sizer=""
        >
          <div
            class="py-6 text-center text-sm"
            cmdk-empty=""
            role="presentation"
          >
            Sem resultados.
          </div>
          <div
            class="overflow-hidden p-1 text-foreground [&_[cmdk-group-heading]]:px-2 [&_[cmdk-group-heading]]:py-1.5 [&_[cmdk-group-heading]]:text-xs [&_[cmdk-group-heading]]:font-medium [&_[cmdk-group-heading]]:text-neutral-500"
            cmdk-group=""
            data-value="Título da Seção"
            hidden=""
            role="presentation"
          >
            <div
              aria-hidden="true"
              cmdk-group-heading=""
              id=":re:"
            >
              Título da Seção
            </div>
            <div
              aria-labelledby=":re:"
              cmdk-group-items=""
              role="group"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
